src/serialization/deserialize-assets.js

```diff
@@ -22,37 +22,29 @@ const deserializeSound = function (sound, runtime, zip, assetFileName) {
         return Promise.resolve(null);
     }
 
-    const assetId = sound.assetId;
-
-    // TODO Is there a faster way to check that this asset
-    // has already been initialized?
-    if (storage.get(assetId)) {
-        // This sound has already been cached.
-        return Promise.resolve(null);
-    }
     if (!zip) { // Zip will not be provided if loading project json from server
         return Promise.resolve(null);
     }
+
     const soundFile = zip.file(fileName);
     if (!soundFile) {
         log.error(`Could not find sound file associated with the ${sound.name} sound.`);
         return Promise.resolve(null);
     }
-    const dataFormat = sound.dataFormat.toLowerCase() === 'mp3' ?
-        storage.DataFormat.MP3 : storage.DataFormat.WAV;
+
     if (!JSZip.support.uint8array) {
         log.error('JSZip uint8array is not supported in this browser.');
         return Promise.resolve(null);
     }
 
-    return soundFile.async('uint8array').then(data => {
-        storage.builtinHelper.cache(
-            storage.AssetType.Sound,
-            dataFormat,
-            data,
-            assetId
-        );
-    });
+    const dataFormat = sound.dataFormat.toLowerCase() === 'mp3' ?
+        storage.DataFormat.MP3 : storage.DataFormat.WAV;
+    return soundFile.async('uint8array').then(data => storage.createAsset(
```

Comment:
So one of the key points of this refactor is here — previously we'd ask `storage` to remember the contents of this asset. Now `storage` just creates it and we return it. The `storage` instance doesn't know about it after creating it.